ch-workflow-superLinter

911samuel opened the pull request:
---

**What does this PR do?**
This PR configures GitHub workflow for linting using Super-Linter.

**Description of Task to be completed**
The task involves configuring Super-Linter for GitHub workflows to ensure code quality and consistency across the project. The Super-Linter badge is added to the README to showcase the integration.

**How should this be manually tested?**
To test this PR manually, follow these steps:
1. Pull the changes from this PR branch into your local environment.
2. Open your project's README file.
3. Verify that the Super-Linter badge has been added to the README with the correct link.
4. Ensure that the GitHub workflow file for linting is correctly configured and includes Super-Linter.
5. Make some changes to your code files to introduce linting errors or inconsistencies.
6. Push the changes to your branch.
7. Wait for the GitHub workflow to run.
8. Check the workflow results to ensure that linting errors are caught and reported correctly.
9. Verify that the Super-Linter badge is updated accordingly based on the workflow status.

**Any background context you want to provide?**
Configuring Super-Linter for GitHub workflows ensures code quality and consistency across the project. Linting helps catch errors, enforce coding standards, and maintain a clean codebase.

**Questions:**
[If you have any questions or need clarifications, list them here]

---

911samuel committed:
configuring super-linter 

Configuring Super-Linter for GitHub workflows ensures code quality and consistency across the project. Linting helps catch errors, enforce coding standards, and maintain a clean codebase.

911samuel commented:
Super-Linter status with a badge in  README:

[![Super-Linter](https://github.com/<OWNER>/<REPOSITORY>/actions/workflows/<WORKFLOW_FILE_NAME>/badge.svg)](https://github.com/marketplace/actions/super-linter)

Emmyfrank commented on .github/workflows/linter.yml:
```diff
@@ -0,0 +1,30 @@
+jobs:
+  build:
+    name: Lint
+    runs-on: ubuntu-latest
```
only runs in ubuntu?

911samuel replied:
thank youfor honest feedback i updated the lint now it is able to run on across all platform and also i added the functionality  to lint only newly pushed changes

911samuel committed:
multiple os and latest pushes

Operating System Matrix:
We're using a matrix strategy to run the job on different operating systems: Ubuntu, Windows, and macOS.
Only Trigger on Push to Main Branch:
The workflow will only be triggered when pushing changes to the main branch.
Check for Changes:
We use a script to determine which files have changed since the last commit. This will provide us with a list of changed files.

Adadie commented:
Consider also adding prettier to ensure the same formatting rules

911samuel committed:
Adding Prittier code formater

911samuel commented:
Thank you for valuable feedback, I updated the workflow to allow prittier code formater

911samuel committed:
Update linter.yml

911samuel committed:
Create .prettierrc
